src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from 'react';
+
+import usersFromServer from '../../api/users';
+import { Todo, TypeError } from '../../types';
+import { getMaxNumberId } from '../../utils/getMaxNumberId';
+import { getUserById } from '../../utils/getUserById';
+
+type Props = {
+  onAddTodo: (newElementTodo: Todo) => void;
+  todos: Todo[];
+};
+
+export const TodoForm: React.FC<Props> = ({ onAddTodo, todos }) => {
+  const [titleError, setTitleError] = useState<string>(TypeError.DEFAULT);
+  const [title, setTitle] = useState('');
+
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedUserIdError, setSelectedUserIdError] = useState<string>(
+    TypeError.DEFAULT,
+  );
+
+  const handleUserInput = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
[CHECKLIST-1] Avoid using one-letter variable names. Instead of 'e', use a more descriptive name such as 'event' or 'inputEvent'. This makes your code more readable.